v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,466 @@
+        } else {
+            throw new IllegalArgumentException("Unsupported type for column " + colName);
+        }
+
+        return date;
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link String} object containing String as value represented in cassandra type.
+     */
+    public static String handleCassandraTextType(String colName, JSONObject valuesJson) {
+        return valuesJson.optString(colName, null); // Get the value or null if the key is not found or the value is null
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link UUID} object containing UUID as value represented in cassandra type.
+     */
+    public static UUID handleCassandraUuidType(String colName, JSONObject valuesJson) {
+        String uuidString = valuesJson.optString(colName, null); // Get the value or null if the key is not found or the value is null
+
+        if (uuidString == null) {
+            return null;
+        }
+
+        return UUID.fromString(uuidString);
+
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Long} object containing Long as value represented in cassandra type.
+     */
+    public static Long handleCassandraBigintType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBigInteger(colName).longValue();
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Integer} object containing Integer as value represented in cassandra type.
+     */
+    public static Integer handleCassandraIntType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBigInteger(colName).intValue();
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link List<Long>} object containing List<Long> as value represented in cassandra type.
+     */
+    public static List<Long> handleInt64ArrayType(String colName, JSONObject valuesJson) {
```

Comment:
Would it make sense to use Generics to handle array types?

For example, in case there's a generic of kind.
`handleArrayType<T> (String columnName, JSONObject valueJson)`
`handleSetType<T>` becomes an extension of the same.

There are various ways to achieve it. One possibility is to have an interface, which has a default implementation for handling sets, based on the overridden implementation for arrays.

Reply:
Addressed the suggestion @VardhanThigle

Reply:
Could you please point me to the commit or change where this is handled?

Reply:
Thanks.